dask/tests/test_distributed.py

```diff
@@ -908,6 +908,76 @@ def test_get_scheduler_with_distributed_active_reset_config(c):
             assert get_scheduler() == c.get
 
 
+@pytest.mark.parametrize(
+    "scheduler, expected_classes",
+    [
+        (None, ("SerializableLock", "SerializableLock", "AcquirerProxy")),
+        ("threads", ("SerializableLock", "SerializableLock", "SerializableLock")),
+        ("processes", ("AcquirerProxy", "AcquirerProxy", "AcquirerProxy")),
+    ],
+)
+def test_get_scheduler_lock(scheduler, expected_classes):
+    from unittest import mock
+
+    da = pytest.importorskip("dask.array", reason="Requires dask.array")
+    db = pytest.importorskip("dask.bag", reason="Requires dask.bag")
+    dd = pytest.importorskip("dask.dataframe", reason="Requires dask.dataframe")
+
+    darr = da.ones((100,))
+    ddf = dd.from_dask_array(darr, columns=["x"])
+    dbag = db.range(100, npartitions=2)
+
+    with mock.patch("distributed.Client") as client_class:
```

Comment:
Why do we need to patch here?

Reply:
Interesting. It appears you're right, I removed the patching. Pretty sure I saw it before not raising, but maybe I was missing something.